### ISSUE TITLE
Documentation of how to wake TV on different subnet

### ISSUE THREAD

**Issue by rkellogg9** [JPersson77/LGTVCompanion#34]
In your guide you mentioned its possible to wake the TV even if its on a different subnet. Mind sharing how??

**Comment by JPersson77** [comment]
Hi, it's not trivial because it requires having network equipment (router or level 3 switch) which can "relay" packet from one separate subnet to another separate subnet. There is tons of information on the internet, but if you try to describe how your network is set up and what hardware you are using, maybe I can help.

**Comment by dechamps** [comment]
If you configure LGTVCompanion to use Unicast ("Device IP address" mode), then *in theory* WoL might work out of the box even if the TV is on a different subnet, as long as the TV is reachable of course (i.e. proper routes are in place).

I say "in theory" because that assumes the TV will respond to ARP requests while off. In my experience with a G1 over WiFi, that is not always the case. YMMV. If the TV doesn't respond to ARP requests then things get more complicated as you will either have to set up a static ARP entry on the last router, or you will need to set up broadcast relaying on every hop (a feature most routers don't even support).

**Comment by rkellogg9** [comment]
I have my C1 connected via ethernet and have a unifi network (unifi dream machine pro and 48 port switch). Currently the LG is on my main lan but i wanted to put it on my IOT VLAN. I have a bunch of firewall rules in place that allow devices like my PC to reach the VLAN. Ill give the static IP address setting a try

**Comment by joebeem** [comment]
I am actually attempting to setup something similar except I have a unifi dream machine standard version. I have vlans set up, but I have not enabled firewall rules to isolate yet (an entire separate issue). However, I  am able to wake the tv on a different subnet without any further tweaking - I just specify the ip and mac address and it worked. I have a feeling when i setup the firewall to isolate, I won't be as lucky.

**Comment by dechamps** [comment]
WoL should work with a firewall as long as it is configured to let LGTVCompanion send packets to the TV on [UDP destination port 9](https://github.com/JPersson77/LGTVCompanion/blob/df0181535bdb9a1b7665826844550d1cc37dc44c/LGTV%20Companion%20Service/Session.cpp#L443). (And for the rest of the control commands it needs Websocket access which happens on [TCP port 3000](https://github.com/JPersson77/LGTVCompanion/blob/df0181535bdb9a1b7665826844550d1cc37dc44c/LGTV%20Companion%20Service/Service.h#L43).)

**Comment by Amoeba00** [comment]
I'm trying to do this exact thing, and while having no issues with the app turning the TV off (just port forwarding tcp 3001) - I can't seem to get the WOL packet to work with all various combos using udp/tcp/both, ports 7,9,3000,3001,9760, and/or 9761.  I know it's not the app, because plugging it into the same LAN, results in both on/off working.

**Comment by JPersson77** [comment]
Thanks  @Amoeba00 for the feedback and the tips that might help more users trying to achieve the same. Port 3001 is for SSL communication, i e the default for LGTV Companion v2.0 and later versions

**Comment by Amoeba00** [comment]
Of course, I just did run into an issue - where if the TV has been off for a while, but computer not asleep and running - then it won't wake up the screen.  Works just fine when on same LAN, but not through the firewall.  If I can figure out what's going on, I'll update here - unless someone has suggestions.  Thanks!

**Comment by yourfriendaaron** [comment]
I have an LG C9 and been struggling to port forward / make firewall rules on Unifi (Cloud Gateway Ultra) for a couple hours now. I forwarded ports 9, 3000, and 3001 from my PC where the Companion is running through to the TV's IP on another VLAN. Not sure what I'm doing wrong.

**Comment by JPersson77** [comment]
Hi, I can confirm that 3000 and 3001 are the ports used for most communication with the TV. If you have that set up correctly you should be able to for example turn the TV off from the test function. 

To be be able to also turn the TV on with the app you need to ensure that wake-on-lan magic packets (according to the method configured in the app) can reach the TV. 

I am no expert on how to set that up though in Unifi so I hope someone else can help you further.

**Comment by Amoeba00** [comment]
Yeah - it's definitely the WOL magic packet that's the issue.  Knowing how to port forward those packets is key to having full control on a different subnet.  Unfortunately, I haven't been able to get that to work.  Of course, if could just be the router I am using that's not able to process those packets properly.

So while you can get the TV to turn off - turning it back on won't work unless you are on the same subnet so the TV can get that packet to wake back up and then receive the controls on 3000-3001 (LG C1 uses 3001).   If I ever figure that part out - I'll certainly share.
